tokenizers/src/pre_tokenizers/byte_level.rs

```diff
@@ -125,13 +124,14 @@ impl ByteLevel {
 // TODO: Give the ability to modify this regex
 impl PreTokenizer for ByteLevel {
     fn pre_tokenize(&self, pretokenized: &mut PreTokenizedString) -> Result<()> {
-        let re_ref: &Regex = self.regex();
-        pretokenized.split(|_, mut normalized| {
-            if self.add_prefix_space && !normalized.get().starts_with(' ') {
-                normalized.prepend(" ");
-            }
-            normalized.split(re_ref, SplitDelimiterBehavior::Isolated)
-        })?;
+        if let Some(re_ref) = self.regex() {
+            pretokenized.split(|_, mut normalized| {
+                if self.add_prefix_space && !normalized.get().starts_with(' ') {
+                    normalized.prepend(" ");
+                }
+                normalized.split(re_ref, SplitDelimiterBehavior::Isolated)
+            })?;
+        };
```

Comment:
I wonder if we don't want the prefix space to be added if we have `regex_type="no_regex"` and `add_prefix_space=True`

Currently I think it will not be added. What do you think about it?

Reply:
You're right. This has been updated.